src/Voron/StorageEnvironment.cs

```diff
@@ -196,14 +199,12 @@ private async Task IdleFlushTimer()
                     {
                         if (SizeOfUnflushedTransactionsInJournalFile != 0)
                             GlobalFlushingBehavior.GlobalFlusher.Value.MaybeFlushEnvironment(this);
-
-                        else if (Journal.Applicator.TotalWrittenButUnsyncedBytes != 0)
-                            SuggestSyncDataFile();
-                    }
-                    else
-                    {
-                        await TimeoutManager.WaitFor(TimeSpan.FromMilliseconds(1000), cancellationToken).ConfigureAwait(false);
                     }
+
+                    if (Journal.Applicator.TotalWrittenButUnsyncedBytes > SyncJournalsMaxSizeInMegabytesThreshold.GetValue(SizeUnit.Bytes))
+                        SuggestSyncDataFile();
```

Comment:
But now we might be suggesting data sync when we have a running write transaction at the same time, don't we? Is it what we want? Note that this is `IdleFlushTimer`

Reply:
Yes, I think this should be inside the if.

Reply:
That would never happen under stress. In fact, it was the reason why the `Sync` would happen without any real heuristic. After every `Flush` we were issuing a `Sync` request that would hammer the IO.